README.md

```diff
@@ -1,66 +1,114 @@
 
-### Deploy
+No further debt can be generated from an ilk which is wiped from `AutoLine`. It also prevents the debt ceiling
+(`line`) for the affected ilk from being changed without Governance interference.
 
-```shell
-$ forge script script/Counter.s.sol:CounterScript --rpc-url <your_rpc_url> --private-key <your_private_key>
-```
+### Set `Clip` breaker
 
-### Cast
+Halts collateral auctions happening in the `Clip` contract belonging to the specified ilks. Sets the breaker level to 3
+to prevent both `kick()`, `redo()` and `take()`.
 
-```shell
-$ cast <subcommand>
-```
+### Disable `DDM`
 
-### Help
+Disables a Direct Deposit Module, preventing further debt from being generated from it.
 
-```shell
-$ forge --help
-$ anvil --help
-$ cast --help
-```
+### Stop `OSM`
+
+Stops the specified Oracle Security Module instances, preventing updates in their price feeds.
+
+## Design
+
+Emergency spells are meant to be as ABI-compatible with regular spells as possible, to allow Governance to reuse any
+existing tools, which will not increase the cognitive burden in an emergency situation.
+
+Previous bespoke emergency spells ([see example][example-emergency-spell]) would perform an open-heart surgery in the
+standard [`DssExec`][dss-exec] contract and include the emergency actions calls in the `schedule` function. This allows
+any contract using the `Mom` architecture ([see example][example-mom]) to bypass the GSM delay.
+
+The same restrictions to regulars spells still apply (i.e.: storage variables are not allowed).
+
+The emergency spells in this repository build on that idea with a few modifications:
+
+1. No [`DssExecLib`][dss-exec-lib] dependency: emergency actions are quite simple by nature, which makes the use of
+   `DssExecLib` superfluous.
+1. No expiration time: contrary to regular spells, which are meant to be cast only once, emergency spells can be reused
+   if needed, so the expiration time is set so far away in the future that in practice the spell does not expire.
+1. No separate [`DssAction`][dss-action]-like contract: regular spells delegate the execution of specific actions to a
+   `DssAction` contract that is deployed by the spell in its constructor. The exact reason for that design choice is
+   unknown to the authors, however we can speculate that the way the spell `tag` is derived<sup>[\[1\]](#fn-1)</sup>
+   requires a separate contract.
+1. Casting is a no-op: while bespoke emergency spells would often conflate emergency actions with non-emergency ones,
+   pre-deployed emergency spells perform only emergency actions, turning `cast()` into a no-op, which exists only for
+   interface-compatibility purposes.
+1. No `MCD_PAUSE` interaction: as its name might suggest, the main goal of `MCD_PAUSE` is to introduce a _pause_ (GSM
+   delay) between the approval of a spell and its execution. Emergency spells by definition bypass the GSM delay, so
+   there is no strong reason to `plan` them in `MCD_PAUSE` as regular spells.
```

Comment:
Unrelated to this code review, I wonder how current "whitelisting"-based monitoring implemented by techops works and if it detects votes on a random non-whitelisted contract before it's been too late / spell is already `schedule`d or if they even depend on the event emitted by `plan` to detect that something is off

Reply:
Let me raise that question to them.

Reply:
From Techops:

> We're "whitelisting" the spells in advance before they are voted through and get the hat. Otherwise any spell that is not whitelisted and that gets the hat, will trigger a P1 alert saying that aunothorized spell has the hat. Where we'll have to react by involving stakeholders and doing necessary actions required for blocking it from being cast.
> Idea is that if that happens and malicious spell is scheduled, we can quickly get all the people online and put "concurrent" spell that would be executed before the malicious one and will cancel it. Or start marketing and informational campaign notifying the users about it. Or preparing some press-release and informing about our next actions. So whitelist is mainly to not any unauthorized spells be executed silently and notifying us about it.